proposed/event-dispatcher.md

```diff
@@ -124,145 +86,94 @@ A Listener Provider is a service object responsible for determining what Listene
 
-## Object composition
-
-A Notifier or Processor SHOULD compose a Listener Provider to determine relevant listeners.  It is RECOMMENDED that a Listener Provider be implemented as a distinct object from the Notifier or Processor but that is NOT REQUIRED.
-
-It is RECOMMENDED that a Notifier and Processor be implemented as distinct objects.  However, an object MAY implement both interfaces if deemed appropriate.
-
-## Interfaces
-
-### General event interfaces
+Listener Providers MUST treat parent types identically to the Event's own type when determining listener applicability.  That is, in the following case:
 
 ```php
-namespace Psr\EventDispatcher;
+class A {}
 
-/**
- * Marker interface indicating an event instance.
- *
- * Event instances may contain zero methods, or as many methods as they
- * want. The interface MUST be implemented, however, to provide type-safety
- * to both listeners as well as listener providers.
- */
-interface EventInterface
-{
-}
-```
+class B extends A {}
 
-```php
-namespace Psr\EventDispatcher;
+$b = new B();
 
-/**
- * Mapper from an event to the listeners that are applicable to that event.
- */
-interface ListenerProviderInterface
-{
-    /**
-     * @param EventInterface $event
-     *   An event for which to return the relevant listeners.
-     * @return iterable[callable]
-     *   An iterable (array, iterator, or generator) of callables.  Each
-     *   callable MUST be type-compatible with $event.
-     */
-    public function getListenersForEvent(EventInterface $event) : iterable;
-}
+function listener(A $event) {}
 ```
 
-### Message Interfaces
+A Listener Provider MUST treat `listener()` as an applicable listener for `$b`, as it is type compatible, unless some other criteria prevents it from doing so.
 
-```php
-namespace Psr\EventDispatcher;
+## Object composition
 
+A Dispatcher SHOULD compose a Listener Provider to determine relevant listeners.  It is RECOMMENDED that a Listener Provider be implemented as a distinct object from the Dispatcher but that is NOT REQUIRED.
 
-/**
- * This is a marker interface used to identify message events.
- */
-interface MessageInterface extends EventInterface
-{
-}
-```
+## Interfaces
 
 ```php
 namespace Psr\EventDispatcher;
 
 /**
- * Defines a notifier for message events.
+ * Defines a dispatcher for events.
  */
-interface MessageNotifierInterface
+interface EventDispatcherInterface
 {
     /**
-     * Notify listeners of a message event.
+     * Provide all listeners with an event to process.
      *
-     * This method MAY act asynchronously.  Callers SHOULD NOT
-     * assume that any action has been taken when this method
-     * returns.
+     * @param object $event
+     *  The object to process.
      *
-     * @param MessageInterface $event
-     *   The event to notify listeners of.
+     * @return object
+     *  The Event that was passed, now modified by listeners.
```

Comment:
Here you indent by one space, below you indent by two. Pick one. (Personally, we use a full indentation - 4 spaces - in such scenarios. Main thing is consistency.)

Reply:
Standardized on 2.  Need to propagate to the package as well.  Mussemfussem...